docs/source/designs/backup-recovery.rst

```diff
@@ -0,0 +1,515 @@
+The convenient-related requirements for the user stories at the top of this design imply that the capabilities for accessing user data will grant access to read the accounting data replicas created by this system.
+This is a strictly worse failure-mode than disclosure of either user data or accounting data separately since it potentially allows identifying information from the payment system to be linked to specific user data.
+Compare:
+* I know Alice has some data but I don't know what that data is.
+* I know someone has dataset **X** but I don't know who.
+* I know Alice has dataset **X**.
+
+This design does not mitigate this risk.
+It may be beneficial to do so in the future.
+
+Backwards Compatibility
+-----------------------
+
+Prior to implementation of this design ZKAPAuthorizer does not maintain backups or replicas.
+Third-parties which have their own backups or replicas should be able to activate the system described here and then discard their backup/replica data.
+
+Performance and Scalability
+---------------------------
+
+Storage Requirements
+~~~~~~~~~~~~~~~~~~~~
+
+We should build a tool to measure the storage requirements of the replica system.
+
+Network Transfers
+~~~~~~~~~~~~~~~~~
+
+We should build a tool to measure data transferred over the network for creation and maintenance of a replica.
+
+Memory Usage
+~~~~~~~~~~~~
+
+We should build a tool to measure memory used by ZKAPAuthorizer with and without replicas enabled so we can compare the incremental cost of replicas.
+
+CPU Usage
+~~~~~~~~~
+
+We should build a tool to measure CPU used by the replica system.
+
+Further Reading
+---------------
+
+* https://litestream.io/
+
+Footnotes
+---------
+
+.. [1] A snapshot is sufficiently up-to-date if the event stream is no more than ``N`` times larger than it.
+       The size requirement exists because the event stream will grow without bounds but the snapshot should have a bounded size.
+       By periodically re-snapshotting and re-starting the event stream the on-grid storage can be bounded as well.
+       Some measurements may be required to choose a good value for ``N``.
+       It may also be necessary to choose whether to prioritize efficient use of network bandwidth or on-grid storage space
+       (and to what degree).
+       If the snapshot does not exist then its size is treated as 0.
+
+.. [2] A snapshot is obsolete if there is a completely uploaded snapshot with a greater sequence number.
+
+.. [3] Application-code is supplied with a cursor which performs this capturing.
+       Replication code bypasses this capturing so that statements which record the event stream are not themselves recorded.
+       Recovery code bypasses this capturing so that statements to recreate the database are also not recorded.
+       ``SELECT`` statements are ignored since they cannot change the database (XXX is this true?).
+
+.. [4] The event stream in the database is large enough when it is larger than 900,000 bytes.
```

Comment:
Is this including or excluding overhead from erasure coding? Can you specify?

Reply:
I'm also curious why 900KB was chosen.

Reply:
I elaborated on this and fixed the doc to describe how to reach the pre-encoding threshold (which is actually 570,000 bytes for our current grid configuration).